$OpenBSD: patch-src_SharedFileList_cpp,v 1.1 2012/02/22 07:54:26 dcoppa Exp $

Fix showing of shared files to other clients
(upstream git commit 4e40aecb160baac72aa28a8a4c7aaf19261e051f)

--- src/SharedFileList.cpp.orig	Sun Sep 18 21:29:33 2011
+++ src/SharedFileList.cpp	Tue Feb 21 15:56:05 2012
@@ -626,10 +626,8 @@ void CSharedFileList::GetSharedFilesByDirectory(const 
 		CKnownFile *cur_file = pos->second;
 
 		if (dir.IsSameDir(cur_file->GetFilePath())) {
-			continue;
+			list.push_back(cur_file);
 		}
-
-		list.push_back(cur_file);
 	}
 }
 
